### ISSUE TITLE
Beta6 Issues & Roadmap

### ISSUE THREAD

**Issue by RowanMaBoot** [Altemio/SevenKingdoms-master#41]
This issue is for our next release, Beta3. Please comment on any bugs and I'll get them added to this list.

If you've any major changes you're working on that you want clearly stated here (you don't need to, but just in case you're working on something that may need manual merging like map changes you may want to).

### Current Roadmap:

From **Rowan/Altemio**
- Introduce a new building system. The building system will be, for the most part, top-heavy with curved bonuses - as you pay more, you gain more. So higher levels will offer considerably larger rewards for construction. Resources on the other hand will be expensive to establish exploitation of, but will get cheaper and more rewarding to construct with each level.
- Regional infrastructure will be handled through a local province modifier rather than a building (I was going to use an event to assign the modifier based on structure construction, but I think it will be possible to use a targeted decision), due to issues encountered with capital switching (since it was capital assigned) and/or issues with effects working as intended.
- Clearly defined structures and requirements, with fewer chained requirements.

From **tsf4**
- Character work: Last bits of 6566 and mostly, new stuff for Age of Heroes.
- Regional Levy buildings, continuing to give them 10 levels and such.
- Mercenaries, continuing to add them for each area of Westeros.
- Validation work

From **TitanRogue**
- a river connecting Greywater Watch with the Green Fork
- bug fixing in general and some localisation (if I get to it); Looks like the Validator is updated so maybe I can figure out whats causing the weird jade dragon event locs
- snowless mountain terrain for the obsidian isles and the iron isllands

### Proposal:
- Use targeted decisions to grow the settlement size so that we may have a combination of prestige, piety and gold costs for growth?

### Known issues to be fixed:
- It has been reported that feudal vassals have negative tax income?
- Provincial revolts are too strong, however most of the time the rebel troops fail to spawn.
- General MP stability (events in general)

**Comment by Vierwood** [comment]
Some issues:

Provincial revolts seem too strong, most of the time however the rebel troops never spawn. 

Castrated boy event spam. 

Lack of good conquest CBs like a generic county conquest CB.

We also need to make MP stable.

**Comment by RowanMaBoot** [comment]
The Castrated event is firing repeatedly? That usually means we've got broken events... MP stability on the other hand is a longer-term project.

**Comment by BladedQueen** [comment]
The ship route bug at The Neck is fixed. I've also added the forested river of Karhold and polished Dreadfort and the upper Green Fork's impassible border. https://imgur.com/a/apEL3lU

Future plans are to add or extend the following rivers; Milkwater, Wendwater, Tumblestone, Red Fork, Blueburn, Cockleswhent. Plus Red Lake. Some may have impassible sections or navigability with province shape adjustment. Feel free to alter any aspect of the map without worry, I can merge my changes around it.

Besides that, 13 unique buildings to add in after finishing the descriptions.

**Comment by TitanRogue** [comment]
@Vierwood 
When does the castrated boy event spam?
It desn't appear in my game.

**Comment by Vierwood** [comment]
@TitanRogue seems very random, but it occurs most often after deciding on an event.

**Comment by RowanMaBoot** [comment]
If you could get specific event examples that'd be great. Try also using debug_mode ingame so you can see event IDs. That likely means something is broken.

**Comment by Vierwood** [comment]
Yeah for some reason the issue only appears in some games. I restarted my single player game and the issue vanished.

**Comment by tannerflick4** [comment]
I have never seen the event.

**Comment by Vierwood** [comment]
We've also  gotten a bunch of reports about it.

**Comment by TitanRogue** [comment]
When did they start? A specific date would help us to figure out what is causing it.

**Comment by Vierwood** [comment]
It's been in the game for a long time, I'm not sure how long though.

**Comment by RowanMaBoot** [comment]
It was reported by players who were playing the mod with an incorrect CK2 version - I'm pretty sure it isn't so much as a problem with the event per se, since it is a totally vanilla event, but more of a problem with events being broken and for some reason that one firing when another event is FUBAR'd.

I've never actually seen it, but on the other hand I constantly crash...

**Comment by Vierwood** [comment]
@Altemio I uploaded some photos for the Workshop, they're a lot better than the current shitty ones.

**Comment by Vierwood** [comment]
Also we should upload a hotfix since I was able to finally fix the law clusterfuck.

**Comment by tannerflick4** [comment]
Which commit was able to fix it?

**Comment by Vierwood** [comment]
The "Bishopric Fix" one. I just went through all of the succession laws and re did them.

**Comment by RowanMaBoot** [comment]
Awwwwesome! I'll see if I can get a release out tomorrow.

What was broken with the laws? Kinda strange since the last commit I did on it was a literal copy-pasta of the vanilla file, so would be good to know what was FUBAR'd!

**Comment by TitanRogue** [comment]
@Vierwood 
I have noticed that some events like favor events have names, description, etc. but are hidden. That doesn't look right. I am not sure, is it intended if yes I'll delete the redondant lines.

This is from 00_tsk_flavor_events.txt
# Knight Minor Title
character_event = {
    id = tsk.0116
    desc = EVTDESCtsk.0116
    picture = GFX_evt_knight
    border = GFX_event_normal_frame_religion'
    hide_window = yes
    is_triggered_only = yes
    
    immediate = {
        if = {
          limit = {
            OR = {
              trait = knight
              trait = hedge_knight
              trait = red_knight
              trait = dragon_knight
            }
          }   
          knight_character_effect = yes
        }
   }

    option = {
        name = EVTOPTAtsk.0116
    }
}

There are events fired by on_action and other events and effects that don't exist, too. That could cause the Castrated Boy event spam.
There are also several dynasties with at least two cultures.

**Comment by RowanMaBoot** [comment]
You'd have to ask @Vierwood about it, since that is his area (he's been adding virtually all of the flavour and EMF events).

There is something I'm worried about in that file tho --- GFX_event_normal_frame_religion'

That rogue ' could cause a whole host of issues.

As for missing events in on_actions, dropping them shouldn't be an issue (generally comment them out in case we bring them back, which is entirely possible as I intend to re-port all vanilla events, since it was a rough/botched job) - however, I'm fairly certain firing a non-existent event doesn't do anything. But maybe it does? Hmm.

**Comment by Vierwood** [comment]
I don't think the hidden events need that stuff, but it doesn't hurt to have them. As for the laws I'm not sure what was broken, but I re did them to fit AGOT rather than Europe.

**Comment by TitanRogue** [comment]
And what about the dynasties?
801076 = {
	name = Greatstar
	culture = bronzeman
}
......
801076 = {
	name = Greatstar
	culture = valeman
}
I know they overwrite each other and are unlikely to cause trouble but we should delete the overwritten lines in order to clean up our code.

**Comment by BladedQueen** [comment]
Red Lake with some farmland and green bog has been added, in addition to a normal height map matching TitanRogue's non-flickering topology map. Red Lake's town position was also changed to be lore-friendly (NE of lake).

Backups of the old topology map and normal height map are in my personal subfolder. I've been experimenting to see if a balance can be struck between preservation of topological detail and reduced flickering.

**Comment by TitanRogue** [comment]
@BladedQueen
I don't understand why an experimental topology map is necessary. Since my edits actually didn't change anything in the game. I simply recolored all (95 95 95) to (96 96 96). That's it. You don't notice it in-game and there is no loss of detail. If I didn't get your point please explain it to me.

**Comment by RowanMaBoot** [comment]
the transition between the old 95 95 95 and 96 96 96 generated a slope which was visible - with all 95 95 95 set to 96 96 96, the transitional regions have been flattened out, resulting in large/(r) 'flat' zones on the map.

BladedQueen provided some screenshots that show a difference and it can be seen, but personally I don't think it is a huge deal.

I'll go over the current topology and will make some tweaks to reduce it.

**Comment by RowanMaBoot** [comment]
@BladedQueen I don't have my normal map tools - only just got GIMP installed, even, so you will want to try to generate a new normal map. It's based on TitanRogue's edits, just with some tweaks to avoid flat-landing.

Very quick, rough and dirty.

**Comment by TitanRogue** [comment]
@BladedQueen @Altemio 
Ok, I somehow missed it.

With your latest commitments you have overwritten all my changes of the province map. Did you make a backup file of the latest province file? So I can merge my edits with yours.

**Comment by BladedQueen** [comment]
@TitanRogue Sorry about that, I mistakenly added an older province map file. All of your improvements have been restored.

About the topology, Rowan explained it, but yes - it led to wide tracts of flat coastal land. It may not seem significant, but it bothered me to lose the variation - which was noticeable in the generated height map. Example: http://www.framecompare.com/image-compare/screenshotcomparison/KWKWNNNX

Your fix was totally effective otherwise at removing all flickering, and I wanted to manually merge your edits into the problematic map areas. It's unnecessary now.

@Altemio New normal height map generated for the revamped topology. I like it. Confirmed that there is no flickering whatsoever in-game, and if anything, there's more topological variation than the older version. Did you only hit Westeros, by the way? I noticed Essos seemed unaffected. :P

**Comment by RowanMaBoot** [comment]
@BladedQueen Awesome! Yeah I've not got to Essos yet, since editing maps is rather arduous.

**Comment by Vierwood** [comment]
@Altemio I did a 250 year sim.

![observer](https://user-images.githubusercontent.com/26224955/43970651-df769a40-9c93-11e8-8e14-2460f53278b8.jpg)

It actually turned out really well. Minus of course the economy of every province being ridiculous.

**Comment by TitanRogue** [comment]
@Vierwood 
It really did! The March government looks well balanced and the other kingdoms look really good as well. I believe we need to strenghten Dorne to resist to foreign invaders maybe by giving the provinces a huge amount of enemy attrition.

**Comment by tannerflick4** [comment]
I added a Mystics society. Everything seems fine with it, except for the interface. You guys can tweak it as you like, but I was basing it off the vanilla hermetics and "mystical" people and stuff in the history/lore for Game of Thrones, such as the search for and use of "magic" in Essos to the underground Westeros goings on like the white walkers, children, and based on how both books and show has many people who you can picture investigating stars, myths, etc.

**Comment by tannerflick4** [comment]
I would like for no one to mess with the cultural buildings. Not trying to sound bossy, I just have an idea of what I want to do with them and don't want interference/balancing done on them yet. 

My mindset is that no longer will the building ("levy") take away from garrison, and each of the 10 levels will only gain a net worth of 250 troops (which is not much at all!). This equals 2500 net worth of troops for a fully upgraded level 10 cultural levy building, which is not bad, especially since it is restricted to capital baronies, AND tied in with prerequisite town/road buildings. It represents more people coming in from the countryside to your castle/keep/city/etc and contributing to your "levy". For example, a Stormlander cultured province would pick up different people for its army than a Northmen province would, due to a difference in the two cultures.

Thank you for holding back the balancing/edits/etc for now, and hopefully you can see the reason I am putting behind these buildings.

**Comment by RowanMaBoot** [comment]
I'm actually working on a building overhaul (so all of the non-cultural buildings will be changing), so please bear it in mind when making your changes! It is quite spreadsheet heavy so far, so it's taking a bit longer to do than it should. The way it is being set up is so that a fully upgraded castle or city may only offer 3,000 troops, at a cost of 16,000 gold and taking upwards of say 150-300 years depending on modifiers, infrastructure and the sort.

The reason for the smaller numbers is because lore-wise an individual barony in many cases could only offer a few hundred troops - less, such in the case of some of the northern houses due to war exhaustion - and having provinces as we do right now supplying 30k+ between each holding is kinda insane.

So I'd be careful how you go about doing cultural buildings - in fact, rather than adding lots of troops I'd look into having them change troop compositions and using them for the retinue size (we can add retinues too) , as well as other small flavour stat tweaks.

Because cultures themselves already do the following:

- Adjust the levy sizes _and_ tax income, to simulate some cultures being more or less fussy about recruits (quantity versus quality).
- Modify the strength of certain unit types, so that they may deal more damage or take less damage during certain combat phases.

But we can't do some other stuff. So for the buildings to work properly, without creating imbalance or redundancy (like just for the sake of being there), they should do the following:

- Change troop compositions. So you'll lose 25 light_infantry but gain 25 archers, for example.
- Instead of creating cultural barracks buildings such as in vanilla, the cultural buildings can be relatively cheap but time consuming to build, working to fill that gap.
- Offer small amounts of tax income or fort level, so cultures known to be harder to invade might see small amounts of bonus fort level.
- Retinue size bonuses?

**Comment by tannerflick4** [comment]
Thanks for the info. I will definitely keep in touch with your building edits, since I really do want to have these tied in with the market, road, village type buildings. 

I will make them deal with troop composition, with fort level and garrison and retinues for the cultures it seems like would make sense to use them with.

I am definitely a building nut, since I enjoy having lots of them to build during a playthrough.

**Comment by TitanRogue** [comment]
Can't wait to see the changes in the game.

**Comment by RowanMaBoot** [comment]
The way the building system is going now is in the direction of bloat avoidance. There will be a few fewer buildings, but each one will matter more. Costs and bonuses will curve either with or against each other where it makes sense - so whilst you might be spending more, you actually get more bang for your buck the more you upgrade a particular tree.

Additionally I want to move building assignment to an event-based system rather than using technology, so we can fine-tune it so higher ranked rulers will have more development.

**Comment by tannerflick4** [comment]
Ok. While I disagree in that direction, I **do** understand it as well. I like the idea of events driving it, but I do not want it to be like the old Middle Earth mod's system, where there was no chance of a different "start" ever. Every time I would play a certain year, it would always start out the same, which I found incredibly boring.

**Comment by RowanMaBoot** [comment]
It will be more dynamic than it is now! For example, every game the Reach and Oldtown is virtually the same development due to the constructed buildings being hard-defined in the history file. Plus or minus a building handled by tech.

Ideally there will be a huge amount of randomisation since we will be able to use proper weighting and random components, and will be able to have more developed seats (for example, if you're a King or Empire tier ruler you will have more development than a count).

Events only control what is initially built - in the same way resources are handed out. If you've the money and interest you will be able to turn any seat into a gargantuan citadel.

Don't get me totally wrong though - tech will determine the bulk of development, but rather than using messy history distribution we'll have buildings assigned via the event dynamically based on your title rank and then secondarily based on certain provinces.

**Comment by tannerflick4** [comment]
Sounds very cool! _heavily anticipates_

**Comment by tannerflick4** [comment]
For some reason, the wildling revolters are calling themselves "Khans" and when they win, it creates a nomadic holding. Very odd, and I cannot figure out how to fix it. Last example I had is a religious revolt for the beyond the wall old gods's heresy, and he was Khagan at the declaration and end of the war.

@Altemio @Vierwood

**Comment by tannerflick4** [comment]
https://forum.paradoxplaza.com/forum/index.php?threads/proof-of-concept-agot-seven-kingdoms.1116399/

What do you all think of this? IMO, I only do not like it because far far in the future, I was planning on doing the book years. Also, seems kind of underhanded, maybe?

**Comment by TitanRogue** [comment]
@tannerflick4 
It´s actually pretty cool. You got to have your stamina bar filled for such a project. But I wouldn`t play it. The map is far too large for these scenarios if you don`t populate it properly.

**Comment by Vierwood** [comment]
If anything one of you guys should approach him and ask if he wants to help us with the mod. We're in dire need of people who are willing to do the history.

**Comment by TitanRogue** [comment]
@Vierwood 
Done.

**Comment by Vierwood** [comment]
Thanks.

**Comment by tannerflick4** [comment]
So are we in for supporting this guy's effort? I am okay with it. It will end up being a mess anyways unless he really keeps up with both team's edits.

**Comment by Vierwood** [comment]
Yeah I don't mind it, I doubt the AGOT team will agree.

**Comment by tannerflick4** [comment]
the new DD is wonderful. This will let people play our mod with random and shattered worlds. Interesting to see how hard it will be to be included

**Comment by RowanMaBoot** [comment]
It wouldn't be feasible for them to support it as a sub-mod, they'd have to re-release our and AGOT's content as an independent, standalone mod.

**Comment by TitanRogue** [comment]
@Altemio 
There are purple and yellow colored columns in the Reach row of the provinceDef.xls. Does it have a function or can I change it to green?

**Comment by tannerflick4** [comment]
no function. it was just made to distinguish the kingdoms when we first started doing it, but tweaks and such since have changed the dejure structure, hence a purple being in reach dejure ;)

**Comment by TitanRogue** [comment]
Good. I put some effort into reorganizing most of it and commited it alongside my last commitment featuring new cultures and province history for Dorne.

**Comment by tannerflick4** [comment]
Great. I will try to update my submod's provDef to match ours.

**Comment by tannerflick4** [comment]
@Altemio @TitanRogue 

What do you guys think of reducing personal demesne? I think kings and dukes have the ability to hold 5+ provinces, and I think with Westeros (compared to real history) we don't see rulers holding vast amounts of land like that. This allows for smaller houses to flourish, instead of being revoked or taken in wars by their liege, which again is more like Westeros. In game, it makes our setup of giving chars 1-3 counties better. 

I know I could play with demesne rule, but I think for "normal" gameplay we need to reduce demense allowed. Perhaps a 50% reduction? King tiers would still be able to hold 2-3 counties, and it would limit counts to 1-2, which is applicable to their supposed size/power in our lore. 

Interested to see what you guys think. I talked with @Vierwood about it on Steam, and he seemed ok with it.

**Comment by TitanRogue** [comment]
@tannerflick4 
I agree, too. All Andal and First Men ruler should be limited to a few holdings. The other rulers like Valyrians, Pentoshi, ... should still be allowed to have a big demesne.

**Comment by tannerflick4** [comment]
Maybe we can 25% reduce Essosi ones and 50% everyone else. Thanks for agreeing :)

**Comment by RowanMaBoot** [comment]
I agree too!

Reduction of demense size is a good idea, plus will mean laws and the sort to boost it will mean a lot more (versus not at all right now).

**Comment by tannerflick4** [comment]
Great! I will adjust it then.

**Comment by tannerflick4** [comment]
Hey guys, @Altemio @Vierwood @TitanRogue 

what do you think about Age of Heroes Essos setup? I would like to make it shattered of course, but didn't Andals control like all of Braavos Lorath and Pentos as well as their homeland? I know before Valyrians they were really strong.

**Comment by Vierwood** [comment]
I'm not sure if the Andals even existed during the Age of Heroes. If they did we might as well go overboard and have playable heroes like Hugor of the Hill, and even the Seven Aspects themselves. 

https://awoiaf.westeros.org/index.php/Hugor_of_the_Hill

**Comment by tannerflick4** [comment]
That was my thought, with his 44 sons and yeah for sure including the aspects and the heresies of the Seven be prominent. It would offer a different playthrough and could be very cool. I'll implement it whenever I get to that area for that bookmark. Maybe the sons could each get a number of counties across the "Andal Empire" (Pentos, Braavos, Andalos and Lorath).

**Comment by tannerflick4** [comment]
Can you guys look at the Order of the Stranger society I added? That, and the Mystics, seem to populate with chars not following the startup_populate rules defined in the files. Please help with this. It is boggling me.

**Comment by TitanRogue** [comment]
They had a nomadic lifestyle at this time so only a few tribal rulers (I suppose along the coastline) existed. I like the idea of the 44 sons, too.
There also were other cultures. Maybe Tyroshi and Myrmen ancestors giving the Andals Braavos, Pentos and Lorath to rule. 44 kings, one father sounds great!!!

**Comment by tannerflick4** [comment]
I almost think I would keep them tribal because it would allow for more options of tribal play. Some kind of way of having each of the sons rule pieces of land in those areas and be tribute to Father would be cool.

Possibly, go the route of trying to add in some republics in places to show the native people. 

It is very exciting to think about ;)

**Comment by TitanRogue** [comment]
Keeping them tribal wouldn't be lore-friendly. We do not have added nomads so far so that be a good addition.

**Comment by tannerflick4** [comment]
Hmm ok. I dislike the way nomads handle land and such but that might just be me.

**Comment by Vierwood** [comment]
I agree, the nomadic government is a nightmare to mod, and you aren't allowed to play as them if you don't own Horse Lords. I think it'd be better if we use tribal.

**Comment by RowanMaBoot** [comment]
You need the Old Gods DLC to play tribal too, unless that changed? I actually sorta like nomads, so I don't think it'd be a bad thing (and settling can be allowed). I mean, we're already taking advantage of lots of DLC-required content anyway.

**Comment by Vierwood** [comment]
@Altemio a guy who is very experienced with event modding approached me and wants to help. His username is: ThePinkPanzer, go ahead and add him as a contributor,

**Comment by TitanRogue** [comment]
@Altemio 
I thought of dynamic title flags and some also in the comments asked for it. What do you think of it?
I made some for the North and tested it. Works well.

**Comment by tannerflick4** [comment]
I think dynamic flags would work well, I mean in lore if you got your house to control the North, you wouldn't be flying the Stark banner! :) 

If its not too hard, then definitely. If you guys want to teach me how to do it, I could work on it as well.

**Comment by TitanRogue** [comment]
It`s actually pretty easy. Ok then, I'll go on.

**Comment by RowanMaBoot** [comment]
@Vierwood 
Great! I've added them as a collab.

Drop them this link: https://github.com/Altemio/SevenKingdoms-master/invitations

@TitanRogue 
I see tanner beat me to it ;) so yeah, sounds like a great feature to me.

**Comment by tannerflick4** [comment]
The beta testers are wailing about the demesne change. Do people even play counts and dukes? 

Anyways, I think I will give an extra 25% for the values, so it will be 75% of the original values.

**Comment by tannerflick4** [comment]
raised_vassal_opinion_loss = no seems to not be working, yet that is what is on the wiki. Any ideas?

Beta testers said they were noticing adding this line to ironborn religions did not change a thing.

Revolts are still... *sighs* not working. Here is a save from our beta tester who says the revolt was leading in "all realm provinces".

@Altemio @Vierwood @TitanRogue 

[weird_revolt.zip](https://github.com/Altemio/SevenKingdoms-master/files/2420376/weird_revolt.zip)

**Comment by RowanMaBoot** [comment]
Which revolts are they? The Wildling, Stepstone and/or religious revolts might still be broken because of scope issues.

**Comment by tannerflick4** [comment]
I at least see some religious revolts work fine. Its like some work and some do not for all those types.

**Comment by TitanRogue** [comment]
@Altemio 
What scope issue are you talking? Do you know how revolts are fired?

**Comment by tannerflick4** [comment]
![20180927180144_1](https://user-images.githubusercontent.com/13467552/46177280-52780580-c280-11e8-87c4-e073cdaefecd.jpg)


So I captured a look at a Wildling revolter, and he is a Nomad and just sits in the province, not actually declaring a war, but holding the CB to do so.

**Comment by RowanMaBoot** [comment]
@TitanRogue 

See `events/04_uprising_event.txt`

The events were modified from vanilla and throw together in less than an hour so they're pretty.. Well, I'm not surprised they don't work, as there is a lot of 'scope depth' and I was basically guessing what level pointed to what. In other words, it was lazy and I did it without really trying to understand what was going on.

**Comment by TitanRogue** [comment]
Thanks.
I am going on vacation today so I can have sleepless nights. I try to fix it.

**Comment by TitanRogue** [comment]
@Altemio 
Do you know where vanilla revolts are fired?

**Comment by tannerflick4** [comment]
old gods rebel events and horse lords

**Comment by TitanRogue** [comment]
When do we update the mod on Steam?

**Comment by tannerflick4** [comment]
We usually agree on a date, I think. And of course make sure its updated/announced everywhere else too.

**Comment by RowanMaBoot** [comment]
We can get a version out next Monday if everyone is OK with that - it’s taking me ages to finish the building overhaul due to work on my end, so it probably won’t be done for then.

**Comment by tannerflick4** [comment]
I am good with that.

**Comment by TitanRogue** [comment]
I would say we wait until the release date of Holy Fury is announced.

**Comment by TitanRogue** [comment]
What do you think of converting from First Men to Andal being only allowed when having an Andal parent?

**Comment by tannerflick4** [comment]
I think that would fit. Andal or Andal type. So like Stormlander and a First men culture char could convert to stormlander.

**Comment by TitanRogue** [comment]
Why is there a House called Ninja and why does it get some land in the Refuge?

**Comment by tannerflick4** [comment]
@Vierwood did the Refuge and its chars, I think.

**Comment by Vierwood** [comment]
It's an Easter egg house.

**Comment by TitanRogue** [comment]
Ok.
About the Isle of Faces... We should make it a wasteland since it is uninhabited and guarded by some green dudes protecting some weirwoodtrees.

**Comment by RowanMaBoot** [comment]
We're actually going to put some children of the forest on it at some point as a sort of easter-egg. And it is a holy site for the old gods, so it's good to have regardless.

**Comment by tannerflick4** [comment]
I did a 200 year run, and we need to reduce the amount of "mass killing" traits being given. lol lots of rulers had it across the observe game.

**Comment by TitanRogue** [comment]
I haven't added any ai_chance so far and the event is triggered on every siege. I am currently working on it.

**Comment by tannerflick4** [comment]
Thanks!

**Comment by TitanRogue** [comment]
I don't feel well about the Valyrians on the map except the Driftmark. I believe it makes much more sense to reduce the amount of provinces of Valyrian culture. Half of it, maybe? The Valyrians built colonies like Lys, Myr, Tyrosh or Volantis but they all got their own culture and so the surrounding settlements. They even killed the dragon lords and their dragons when they rallied in Tyrosh after the Doom. I suggest we split the Lysene culture in Lysene and maybe Western Valyrian so the southeastern provinces are Western Valyrian (or whatever you want to call it) and the rest Lysene. I am concerned about lore-friendlity (?) and when making up my mind about how the Doom could take place in our mod I started to think about it.

**Comment by tannerflick4** [comment]
I have made Lys occupied by Valyrians because it is necessary for the Viceroy (for some reason) to have land otherwise the ability to interact vanishes with the viceroy. 

That said, I'd be ok with making a western valyrian. It could be halfed with Lysene too.

**Comment by TitanRogue** [comment]
@Altemio 
I have some suggesions for map in coming bookmarks like "The Doom of Valyria wich takes place around 114 years before Aegon's Conquest according to awoiaf. So at this date the map should have more sililarities with the map we now from the books. So these are my suggestions:
![map-suggestions](https://user-images.githubusercontent.com/40209082/46796139-32e6df80-cd4c-11e8-86bd-0620c501a36a.png)
Green provinces are moved or/and rescaled. Red represents their former position. There would be no loss of provinces/titles and I believe that we could make bookmarks in the early history of Westeros with a lore-friendlier map.

I deleted the minor islands around the Smoulering Isle by mistake.

**Comment by tannerflick4** [comment]
The only thing is, this would be changed for the whole timeline. I think as long as we dont get rid of titles we will be fine though. I am all for this.

**Comment by RowanMaBoot** [comment]
I'm not too big about removing/moving/resizing the islands and similar for two reasons:

- There isn't a whole lot that is lore unfriendly about the map, so it's hard even for purists to complain. Sure, GRRM's map has virtually no detail at all and therefore doesn't include a lot of the stuff we've added. But it was made and tailored for his books, not for a game. Even the TV show changed a lot of things geographically (and apparently portrayed stuff in some cases, like the regions around Kings Landing, more accurately to his vision even if they don't come across even remotely like that in writing).
- It's a pain in the arse to do, because it involves editing the province.bmp, terrain.bmp, trees.bmp, heightmap & colourmaps.

I'm not at all adverse to the proposed change in the Iron Isles for example, since I've never been a fan of that big island.

But the changes to the isles around the Arbor, Dragonstone and Tarth might be a bit much - at those sizes we'd be better off removing the provinces/titles, since they'd be getting too small for proper play (and I'm not sure they'd be worth the effort, since they're hardly taking away from gameplay).

**Comment by Vierwood** [comment]
I agree, if we're going to remove any islands it should be the one in the Iron Islands. The other ones are fine.

**Comment by tannerflick4** [comment]
Well, I don't mind what we remove, as long as its heavily detailed on what we do remove and/or change. I would say, we can always add those province counts somewhere in the nearby region. For instance, making a larger Dragonstone with it being 1-3 more provinces, same for Arbor and Tarth, and iron isles too.

**Comment by TitanRogue** [comment]
Ok, so you all agree with the Iron Islands. Can I go ahead and edit terrain, trees, river and colormaps? I am not that good with making detailed topolgy maps so I would leave that to you.
There are some Provinces in the definition file wich have no colors on the province map. If you agree we could make them sea provinces.

**Comment by RowanMaBoot** [comment]
I've no problem with that! I'll be working on the mod proper this weekend, as I'm actually off - so I should be able to touch up the topology and the sort.

**Comment by TitanRogue** [comment]
@Altemio 
Could you also take a look at the Torrentine. The mountains are kinda glitching:
![cdkned](https://user-images.githubusercontent.com/40209082/46883160-6c554300-ce51-11e8-8fb3-03bb46a2862b.png)
And what about the empty provinces?

**Comment by RowanMaBoot** [comment]
What exactly is glitching - the river edges? And which empty provinces?

**Comment by TitanRogue** [comment]
@Altemio 
One of the beta testers posted his error log:
[map.cpp:1449]: Province 872 has no pixels in provinces.bmp
[map.cpp:1449]: Province 873 has no pixels in provinces.bmp
[map.cpp:1449]: Province 874 has no pixels in provinces.bmp
[map.cpp:1449]: Province 875 has no pixels in provinces.bmp
[map.cpp:1449]: Province 1290 has no pixels in provinces.bmp
[map.cpp:1449]: Province 1295 has no pixels in provinces.bmp
[map.cpp:1449]: Province 1296 has no pixels in provinces.bmp
[map.cpp:1449]: Province 1355 has no pixels in provinces.bmp
[map.cpp:1449]: Province 1356 has no pixels in provinces.bmp
[map.cpp:1449]: Province 1480 has no pixels in provinces.bmp
[map.cpp:1449]: Province 1481 has no pixels in provinces.bmp
[map.cpp:1449]: Province 1482 has no pixels in provinces.bmp
[map.cpp:1449]: Province 1483 has no pixels in provinces.bmp
[map.cpp:1449]: Province 1485 has no pixels in provinces.bmp
[map.cpp:1449]: Province 1486 has no pixels in provinces.bmp
[map.cpp:1449]: Province 1488 has no pixels in provinces.bmp
[map.cpp:1449]: Province 1489 has no pixels in provinces.bmp
[map.cpp:1449]: Province 1490 has no pixels in provinces.bmp
We could use some more sea provinces, I believe.
Yes, the river edges but never mind it is no major issue.

**Comment by RowanMaBoot** [comment]
Ah gotcha, I think that is due to how that part of the river is defined (it is either a wasteland or undefined province). And yeah more sea provinces will probably be a good idea, especially in regions with higher island and coastal province density.

I'll check out the rest of those provinces - curious to know which provinces they are meant to be.

**Comment by TitanRogue** [comment]
The province's colors are mostly deep black tones like 1;1;2 or 50;1;1. And there is only one wasteland beyond the wall and I checked some of these colors. They don't appear in the provinces map.

**Comment by tannerflick4** [comment]
Hey guys, I still am around :) Just been busy these past days, hope to remedy that soon !

**Comment by RowanMaBoot** [comment]
I've been busy as heck for a while, so I'm in the same boat arrgh. Happy to hear though :)

**Comment by tannerflick4** [comment]
Whats our plans for Holy Fury? Just work on compatching and the new features as we can?

**Comment by TitanRogue** [comment]
The bloodline feature, obviously. I want to allow close-kin marriages between Valyrians only after the Doom of Valyria. With the new reform religion system it might be possible to change a religions doctrines via commands and events.
Looks like there is to be an Duel Engine to be implemented by PDX. If so and if it is better than EMF's I will use it for the Dragon Engine and also replace our current one.
The new Crusades can be adjusted to fit the Andal Invasion. Andal Lords could be favored for the targetted title.
The new succession laws will be great for the Blackwater and especially Valyria.

I think we should release the next Beta as soon as possible if everyone agrees. There is lots of stuff to adjust when Holy Fury is released and it will take quite some time, I fear.

**Comment by RowanMaBoot** [comment]
Just let me know when we want to put it up, and I'll stick it up - I'm off tomorrow so will go over some stuff that needs to be touched up on.

In case anyone has been wondering about my absence, I've been very busy with work and working on the mod kinda feels like work, so I'm saving my energy for the Holy Fury release - since a lot will need to be done, especially from my end.

Additionally, the mod crashes constantly for me (like as is reported by many players) and I'm not yet sure, so I've got to do some rigorous testing to see if I can determine what the cause is.

**Comment by tannerflick4** [comment]
Might as well post it up if you can fix your crashes or find the cause.

Yesterday I did some bugfixing and made Qhored the Cruel "own" the entire Isles by vassalizing everyone (of course I will eventually make regular families and such for that bookmark).

**Comment by TitanRogue** [comment]
I read an article wich claims that Holy Fury will contain dragon portraits. Great!!!

**Comment by TitanRogue** [comment]
Also I suggest that we do a comprehensive remake of al events (I know there are many). That should fix some broken events and crashes. It would be enough to compare our events with the vanilla ones. Since there are to be many changes with Holy Fury taking a closer look to the events is necessary.

**Comment by tannerflick4** [comment]
I definitely plan on looking closely on our events vs vanilla ones.

**Comment by TitanRogue** [comment]
I will support you. I will also take a closer look on the gfx and interface stuff.

**Comment by RowanMaBoot** [comment]
Yeah, my main objective is to effectively re-import all of the events where possible.

A while back the editor I was using at work broke a lot of files, because of a setting (replace tabs with spaces on file load) that will make diff compares an absolute pain, so it may not be feasible to just compare the differences.

**Comment by TitanRogue** [comment]
Patch 3.0 is out and the game crashes, unfortunately.

**Comment by TitanRogue** [comment]
In order to avoid overlaps we should divide the new features among each other.

**Comment by RowanMaBoot** [comment]
I'm planning on a lot of trait and event work, so I can pick up and deal with anything in regards to that. 

Wouldn't worry about too much overlap in that regard - but I'm planning on effectively re-importing and re-editing all vanilla events. I won't touch anything if someone has done any work on a particular event file, and will try to bring in any changes that might have been done by anyone in the team in the past.

**Comment by Vierwood** [comment]
The portraits and portrait properties got changed drastically. It'll take me a awhile to go through it all.

**Comment by TitanRogue** [comment]
Did someone get it running?

**Comment by RowanMaBoot** [comment]
I'll be working on it this weekend and over the course of the following (for me four day weekend), so should be able to get it running then.

**Comment by TitanRogue** [comment]
I suggest we change offmap_freehold to offmap_china again since replacing every line with freehold is kinda irritating.

**Comment by RowanMaBoot** [comment]
Agreed, I'm also tempted to do the same with religions..

**Comment by TitanRogue** [comment]
Are you sure? The china freehold thing is easy to remember. But giving vanilla religions a new name... I think we would lose track.

**Comment by RowanMaBoot** [comment]
Not sure about it, but an idea that has come to mind.

**Comment by TitanRogue** [comment]
Small progress update: the randomize and shatter world feature does now work without any crashes.
But starting the game does still not work.

Should we now completely delete the vanilla events and decisions or shall we wait until we got it running with 3.0 and deleted them afterwards?

There is a mod called "Bigger Interface", what about implementing it in our own?

**Comment by TitanRogue** [comment]
The Seven Kingdoms in combination with "In heavens Graphics overhaul":
![20181212221206_1](https://user-images.githubusercontent.com/40209082/49902033-20c81100-fe63-11e8-87e3-d1b6d17f3a69.jpg)
![20181212221231_1](https://user-images.githubusercontent.com/40209082/49902034-2160a780-fe63-11e8-8a98-56cfd8e2f0c9.jpg)
What do you think of it?

**Comment by TitanRogue** [comment]
![20181213170521_1](https://user-images.githubusercontent.com/40209082/49951042-53244d80-fef9-11e8-99ac-9a9b5eb54315.jpg)
It works !!!!

**Comment by tannerflick4** [comment]
So glad to see progress on updating being made, thanks to you guys. I have been busy, and will definitely get back to the mod. Keep on it! :)

**Comment by RowanMaBoot** [comment]
Those textures are quite a bit nicer than mine... To say the least. Beyond that though, excellent work so far everyone! Ya'll making me feel lazy - I'm trying to get some stuff pushed and fixed right now, although I seem to be breaking more than I'm fixing.

**Comment by TitanRogue** [comment]
@Altemio 
Thank you. Unfortunately your right. Since your last commitments the game crashes on loading screen.

**Comment by RowanMaBoot** [comment]
Strange, I broke it with one of the Government commits, but the quickfix commit fixed crashing on my end. Is it still an issue post event work?

Edit: Oops. I see - the offmap freehold file. I completely missed that one - good catch!

**Comment by tannerflick4** [comment]
Reverted my update work from yesterday, Titan said it crashed the mod, and I was sick so mind was kinda foggy. @Altemio , shouldn't the Chinese code from vanilla which we have as "China" be changed?

**Comment by RowanMaBoot** [comment]
Yeah, anywhere that is down as 'chinese' in vanilla should be modified to say chinese in our mod.

Basically I did a blanket search and replace for all references of freehold to china, so naturally if chinese anywhere was changed to freehold it will now say 'china' instead, which won't be what we want.

**Comment by tannerflick4** [comment]
@Vierwood why were my edits from today reverted? Huh? It was the start of fixing the China stuff Rowan talked about right above this comment...

**Comment by Vierwood** [comment]
It was causing a major crash upon game start.

**Comment by tannerflick4** [comment]
ok, well, I guess I won't do the china to chinese fixes then. Hopefully someone else who has more talent than me can try, cause clearly right now I cannot. Not trying to be rude, but I just feel like lately anything I do causes problems. I will stick with something else with the mod.

**Comment by TitanRogue** [comment]
I'll take care of it. I also suggest that we sort all characters by their birth dates and dynasties.

**Comment by tannerflick4** [comment]
How is the general 3.0.x updating going? What else needs to be done?

**Comment by TitanRogue** [comment]
-Bloodline balancing
-Getting alternate start running
-Events still cause crashes so...
-Decisions work
That is what comes up to my mind. I might have missed something.

**Comment by RowanMaBoot** [comment]
Chipping away at events very slowly. Editing them manually this time around (so going through tens of thousands of lines, so its taking an hour or more alone to deal with some files), rather than using global search and replace functions to get things working quickly - should mean less work down the line, less crashing and less broken stuff.

Let me know if crashing ceases at any point!

**Comment by Vierwood** [comment]
@Altemio thanks for all the work!

It seems that at least half of the errors in the log are due to the mod lacking many of the Holy Fury scripted triggers.

I also noticed a few of the errors had to do with EMF, so that might be causing the crashes.

**Comment by RowanMaBoot** [comment]
Thanks to everyone for the work put in so far! Hopefully we should be able to get it working some time over the next few weeks.

Scripted triggers will most certainly contribute to troubles! I don't have access to the vanilla files at the moment, so I'm not in a position to 'restore' the HF scripted triggers and/or effects.

**Comment by TitanRogue** [comment]
It is a bit early in the morning but I already found a missing bracket in the scripted triggers file. I share your hopes Rowan/Altemio.

**Comment by Vierwood** [comment]
About 9,000 of the 12,000 errors are being caused by the scripted triggers and effects...

**Comment by tannerflick4** [comment]
Ok, just did an observe for 25 years and it was still going strong. Great work on this guys, and keep it up!

**Comment by tannerflick4** [comment]
Alright, so playing as an Andal lord in 5200s it only went 6 days :(

The council actions were all non existent besides the collect taxes option for stewards.

![20190223155257_1](https://user-images.githubusercontent.com/13467552/53291615-c3faa800-377b-11e9-8e51-b7d720ad2913.jpg)

**Comment by RowanMaBoot** [comment]
Great! Lot to do still! Good to know it's able to survive longer now with all the work we've been doing. Good sign that we're on track - just taking a while to go through the events.

Should be way more stable down the line I hope. It's kind of a first pass right now, modifying all of the events to work with the mod - further tweaking and balancing will obviously need to be done down the line.

**Comment by Vierwood** [comment]
@Altemio the broken scripted effect is "random_coronation_events_effect." It's causing all of the effects after it to also not work. 

I've tried to find the issue with it, but it's rather large and complicated. Could you take a look?

**Comment by TitanRogue** [comment]
Ok, fixed all job actions except the diplo stuff.

Update: All jobs are fixed.
I also found the missing bracket in scrpited_effects.

Character 70116 is duplicate: Line of Ulrick and Line of Dagos
There are also many validator entries about two holder per start of reign date. You can check this by validating title history.

**Comment by Vierwood** [comment]
Just thought I'd mention a small issue. The crossing at Fair Crossing is not working properly.

**Comment by HungarianScienceAcademy** [comment]
How is the Beta3 update going?

**Comment by TitanRogue** [comment]
@Vierwood Did not noticed that. I'll take a look at it.
@HungarianScienceAcademy So far good.
I suggest that we remove all EMF files from our code. This way we can release the update much faster.

**Comment by tannerflick4** [comment]
I would like to do the updating for 3.1, as I took care to backup the game files so I know what the devs changed, and am updating other mods as well.

EDIT: Ok so @TitanRogue has done the common folder updates. If anyone else has started doing the rest, please let me know. Its ok, we just don't need all of us working on the updating at the same time ;)

**Comment by RowanMaBoot** [comment]
I'm chipping away at events, but all of the unmodified event files are from Holy Fury - and the modified ones are modified _from_ Holy Fury. So events will need a second pass eventually.

Also double reminder to comment out 'removals' rather than actually removing the lines in question, as it makes comparisons and merges a lot easier. ;)

**Comment by TitanRogue** [comment]
Sorry guys, my email is broken. Did not see that there was a conversation.
However, what do you think of making winterfell, the eyrie, casterly rock,...great works? We did also miss some of the 3.0 lines in scripted_effects and scripted_triggers but they should now be completely merged. I apologize for redundant work and waisted time I might have caused.
If get enough time in the next week. We might be able to make the Dragon Engine part of our next update. But with a game rule since it still is in WIP, especially the dragon duells.

**Comment by tannerflick4** [comment]
I am all for making those into great works 👍

**Comment by TitanRogue** [comment]
Shall we split the regions for the great works stuff?
If so I would like to take care of the north and the vale.

**Comment by tannerflick4** [comment]
I would say, go ahead and "take" a region and if you want another, then come back here and say you are going to do another?

**Comment by TitanRogue** [comment]
Okay I have finished the Eyrie, Winterfell and Storm's End. Next ones should be Casterly Rock and Oldtown (Hightower) as preexisting great works.

Update: Hightower and Casterly Rock are added as well. The last great work that comes to my mind is the Titan of Braavos. If you have any other great works that should be added let me know.

**Comment by ZeroFighterR** [comment]
The Wall is the only one coming to my mind 


(And I'm not dead, just have no time to dedicate to help with my full-time job)

**Comment by DeveloperAmr** [comment]
> 
> 
> Okay I have finished the Eyrie, Winterfell and Storm's End. Next ones should be Casterly Rock and Oldtown (Hightower) as preexisting great works.
> 
> Update: Hightower and Casterly Rock are added as well. The last great work that comes to my mind is the Titan of Braavos. If you have any other great works that should be added let me know.

I think the iron islands could have some sort of great work, however, I am unsure what could be considered a great work since not much has been talked about them

**Comment by Vierwood** [comment]
For the Iron Islands I'd do "Nagga's Hill."

https://awoiaf.westeros.org/index.php/Nagga%27s_Hill

**Comment by tannerflick4** [comment]
Ten Towers is actually a great idea for an Ironborn great work, since it was one of the largest and was an extensive project over time to build.

**Comment by Vierwood** [comment]
> Ten Towers is actually a great idea for an Ironborn great work, since it was one of the largest and was an extensive project over time to build.

I don't think Ten Towers has been constructed yet for our bookmarks. We could however make it buildable.

**Comment by tannerflick4** [comment]
So, after taking a look at the alternate start, I have been able to connect islands and BTW to the mainland, down to the Iron Islands. If someone would like to pick up the work, PM me on Steam or discord. I will keep picking away at it, but cannot devote much time to it. The previous adjacencies were not letting the "spread" work. Also, if you know how to make sure there is no "noculture" instances on map that would be great. they show up randomly each time a alt start is generated.

**Comment by tannerflick4** [comment]
@Altemio @TitanRogue 

Getting this error, and not sure why, as wildling government does exist...

`# Other Governments
is_special_other_government_potential = {
    OR = {
        is_government_potential = wildling_government
        # is_government_potential = high_tribal_government
    }
}`

`[triggerimplementation.cpp:17910]: Script Assert! assert: "false && "CIsGovernmentPotentialTrigger has wrong Scope"", type: "none", location: " file: common/scripted_triggers/tsk_government_scripted_triggers.txt line: 33"`

**Comment by TitanRogue** [comment]
@tannerflick4 
Looks like wildling_government doesn`t exist.

**Comment by TitanRogue** [comment]
@Altemio @tannerflick4  @Vierwood 
Which coronations shall we use? I would prefer EMF since they have the cool invitations.

I still have some struggle finding suitable portraits for our dragons and today I found this mod on Steam:https://steamcommunity.com/sharedfiles/filedetails/?id=1733862215. I believe this is a opportunity for us to have major diffrence between TSK and AGOT since right some portraits are used by both of us. What do you think of the portraits? Shall we use them?

**Comment by tannerflick4** [comment]
@Altemio , this I think is the best place to find you on :) I am lead dev on the MEP team, and my other team members wanted to know if you would be interested in redoing the Middle Earth map for us?

**Comment by RowanMaBoot** [comment]
> 
> 
> @Altemio , this I think is the best place to find you on :) I am lead dev on the MEP team, and my other team members wanted to know if you would be interested in redoing the Middle Earth map for us?

I could probably do that fairly easily! What exactly would you be looking for me to do?

@TitanRogue 

I'd also prefer EMF for it too!

As for the portraits, I think they look pretty good - feel free to ping them to find out if we could use them too (I can't open the link right now, at the office, but if it's the one I think it is I'm pretty sure they themselves are using someone elses' assets).

**Comment by tannerflick4** [comment]
@Altemio what is best way for him to get in touch with you?

**Comment by tannerflick4** [comment]
@Altemio , whats the best way for Marowa on our MEP team to contact you?

**Comment by TitanRogue** [comment]
@Altemio @Vierwood @tannerflick4 
What do you think of using the byzantine government and succession laws for the Western Territories?

**Comment by tannerflick4** [comment]
I am good with that. Does the Viceroy even need to be a republic now that I assume we can make the free cities as an "ok" gov type as a vasssal?

**Comment by TitanRogue** [comment]
@Altemio @Vierwood @tannerflick4 
What do you think of using the byzantine government and succession laws for the Western Territories?

**Comment by Vierwood** [comment]
I wouldn't mind it. Although, we'd have to ensure that the Valyrian Territories don't get any ideas about invading Westeros.

**Comment by TitanRogue** [comment]
About event patching: All events except for old_gods_various_events, old_gods_holmgang, mnm_misc_events, mnm_artifacts_events, misc_old_gods_events, HF_sainthood_events, HF_religious_events, HF_bloodline_events, HF_alternate_start_events are done.

**Comment by RowanMaBoot** [comment]
@TitanRogue 

Would it make more sense for the Western Empire to not be ruled by a republic of Valyrian Dynasties? We've been calling the individual in question that has been elected to rule over the subject republics a viceroy - but would having them actually function as a (feudal) viceroy even be more 'realistic' to the canon?

**Comment by Vierwood** [comment]
A lot of the Jade Dragon "China" localisation has reverted back to vanilla, causing words such as China and Emperor to be used rather than Valyria and Archon.

**Comment by TitanRogue** [comment]
@Vierwood I will edit it in the coming days.

@Altemio You are propably right, it should stay a republic-like government. Some people on Steam complained about the GitHub version not being linked in the description. Could you add a link for it and also for our Discord server?

**Comment by RowanMaBoot** [comment]
Per Paradox we can’t link the github repository (due to people using external links to circumvent needing to own the game to get mods). Will look into adding the discord link once I’m able to (feel free to link Git there).

(In case anyone is curious about my disappearance, I moved house (currently no internet there, waiting for charter to finish burying cables), got a new job/promotion and I am currently on vacation out of state so I’ve no ability to work on the mod atm. Lot happened in a very short period of time, so still getting my ducks lined up)

**Comment by tannerflick4** [comment]
@Altemio , whats best way for Marowa on the MEP team to contact you about the map?

**Comment by GoogleDavide** [comment]
if this is the correct place to report bugs, then I wanted to report 2 things
Women don't lose the pregnancy trait after giving birth to a stillborn, so they can't give birth to other child
When starting a saved game, all yours personal levies go down to zero
All of these appened in the Age of Heroes bookmark
(if this isn't the right place to post such things, let me know and I'll remove my comment, btw thanks for the amazing mod!)

**Comment by RowanMaBoot** [comment]
> 
> 
> @Altemio , whats best way for Marowa on the MEP team to contact you about the map?

Should be good to find me on the Paradox forums (I'm not particularly active on there, but if you let me know that they've reached out to me on there I'll get onto it). Easy to share stuff too on there as well!

**Comment by tannerflick4** [comment]
@Vierwood , whats your idea on the climate? Beyond the wall gets severe winter, and the north gets normal, nothing for the rest? I am trying to get the ProvDef back in sync...

**Comment by Vierwood** [comment]
@tannerflick4 that's the idea, yeah.

**Comment by tannerflick4** [comment]
ok, I'll change it in the ProvDef to reflect this tomorrow.

**Comment by tannerflick4** [comment]
Ok, at one time we had a bunch of the JD/China code really set up for our mod, and now it seems like its been put back to vanilla in the events. Why was this done? It was quite a lot of legwork getting everything changed over and accommodating the extra status...

**Comment by RowanMaBoot** [comment]
If you're talking about events, when HF came through I ported over every single event from vanilla from scratch.

This is because we were getting crashes due to a lot of broken events (ranging from variables that were renamed in some places but not in others - we shouldn't have renamed any of them - to regions, provinces, cultures and religions that were outright replaced or removed, resulting in broken triggers -- and so on so forth) from a large portion of the files in the mod.

When we originally 'ported' them, it was a case of brute force over manual, fine-tuned edits and tweaks. It meant we couldn't do compare and replace of our events versus new vanilla events, since so much was changed. Plus I sort 'broke' many of the files due to the editor I was using replacing tabs automatically with spaces (meant every single line was flagged as changed, made comparison merges impossible for many files).

The old events should still be in a subfolder, so you should be able to do compares against some of the old events versus the new ones, as not all of them were affected (just most).

But in regards to backbone, we need to keep as much as 'vanilla' as possible to make future updates painless. In other words, avoid renaming variables and the sort because new events will use the old variables etc (and they're just variable names - no actual reason to change them). If that makes sense.

**Comment by tannerflick4** [comment]
Yeah, the whole China/Chinese was a big mess, because we tried to change it, and then bring it back.

Thank you for the explanation Rowan. I thought it might of been you, but thats ok, we didn't do alot of that the right way because someone dove in and was drowning so we helped them out, but it got messy.

I will look into the China/Chinese conundrum, and try to make it work seamlessly for Valyria, which will add a lot of flavor and change up Essos. When I worked on the invasion events, I tried but failed to really do a "3rd" invasion (Ghiscari) modeled after the China/Mongol invasion, but couldn't get it done. So I am sure this led to errors. Of course, any advice or comments on it as it happens can be directed to me, I don't mind explaining X or Y change.

**Comment by TitanRogue** [comment]
@Altemio 

Could you please link our Discord server at Steam

**Comment by TitanRogue** [comment]
Also I would be happy if some native English speaker could take a look at my localisation files for custom mechanics I have uploaded recently.

**Comment by RowanMaBoot** [comment]
> 
> 
> @Altemio
> 
> Could you please link our Discord server at Steam

Seems Steam censors Discord links for some reason..?

**Comment by TitanRogue** [comment]
@Altemio At least in the comment section

**Comment by TitanRogue** [comment]
There is this mod on Steam called "Improved Genetics 2.0" which adds a gene-tracker (?). In order to represent incest properly a mechanic must be found or developed that keeps track of a character's parents and evaluates possibly incest traits. This mod might be a good choice as close family members have a high chance of producing clubfooted, hunchbacked, etc. children. If you all I agree I might ask him for permission to integrate it into TSK.

**Comment by tannerflick4** [comment]
Sounds great to me. Count me in :)

**Comment by RowanMaBoot** [comment]
Yeah that'd be awesome, I've personally used that mod with HIP and it is quite subtle, to say the least.

**Comment by TitanRogue** [comment]
We have some congenital traits added to the game which might be lost if we add Improved Genetics. I could try to integrate them, but I do not think that they are necessary. We will get some new traits in return though.

**Comment by TitanRogue** [comment]
I also ran into an issue:
After a few years some characters get a hidden trait "cannot_marry" wich is used by several vanilla events to manage scripted marriages or to prevent characters from marrying for a specific amount of time. However, it seems like this trait does not get removed.

**Comment by tannerflick4** [comment]
So, a modding friend of mine found an issue with the ProvDef tool, and is going to try to fix it. It makes all ocean provs into navigable rivers, which means only our cultures which can navigate can use the oceans. Problematic, but should be an easy fix he says.

**Comment by TitanRogue** [comment]
This issue does not occur in our mod, does it? I can navigate through all oceans I want without any issues but not through navigable rivers as a Stormlander ruler.

**Comment by tannerflick4** [comment]
I am not sure, but my friend thinks someone must of already figured this out for us.

**Comment by RowanMaBoot** [comment]
I've not seen this issue with the ProvDef tool, so if a newer release has been pushed it could be an issue with that version of the tool - no for me to know off-hand.

However, so long as the provinces are set up properly in the spreadsheet I don't think we've ever had issues with them being assigned to ocean/rivers etc. incorrectly.

Assignments are handled via the default.map file, where rivers are assigned the the major_rivers block and ocean provinces assigned to the sea_zones blocks (which serves as range identification. ID of province x to x, including all between)

**Comment by TitanRogue** [comment]
@Altemio @tannerflick4  @Vierwood 
Why do we need the far east religion group?

**Comment by tannerflick4** [comment]
It was intended to show the small chance rulers had to convert to foreign religions, like Stannis in the books with Rhllor. Essentially, we were hoping for a intricate system which would see courtiers pop up in courts with a small chance of initiating an event to convert the ruler.

**Comment by TitanRogue** [comment]
Actually reminds me of an IR event when a group of egyptians arrives in one of my cities and converts some pops to their orientalic religion and culture.

**Comment by TitanRogue** [comment]
I suggest we give the Rhoynar and Dothraki their own religion group. Since they actually do not have much in common.
I was also thinking about the passage connecting the Reach and Dorne (Carrion Pass). I think unless PDX does not add movement blocking forts this passage should be blocked by wasteland.

**Comment by tannerflick4** [comment]
They were just grouped like that for convenience, since they would only come through Valyria offmap anyways. But go for it! Just be sure to switch/add the references, they are in quite a few places.

**Comment by TitanRogue** [comment]
Duplicate character IDs:
7125, 7126

**Comment by TitanRogue** [comment]
@Altemio  @Vierwood  @tannerflick4 
I might have found a way to simulate the Bloody Gate and Moat Cailin. There is a way to turn wastelands into actual provinces with holdings. Whenever an enemy army is in the neighbouring province the Bloody Gate turns into wasteland and is impassable and opens only when the enemy army leaves the provinces or an allied host is on the other side.

**Comment by tannerflick4** [comment]
Interesting, wouldn't it be best to just give like a big negative modifier for anyone who attacks them?

**Comment by tannerflick4** [comment]
@Altemio , please message me on Steam. A group of myself and some modders are interested in seeing if you would like to help us mapping wise.

**Comment by TitanRogue** [comment]
@Altemio 
I propose we release our current GitHub version on Steam. They have been waiting for almost a year and the version is in a good shape.

**Comment by TitanRogue** [comment]
What shall we do with the Western Territories? Should they have their special Freehold government again, confucian bureaucracy (renamed) or something else?

**Comment by TitanRogue** [comment]
@Altemio I also ask to allow me to upload the Steam release from now on. I could push smaller updates much more frequently. Do not get me wrong, I do not want to undermine you. ;)

**Comment by tannerflick4** [comment]
I would go with what is mentioned in the JD event files, then you wouldn't have to change every instance of it? But I really am not sure.

**Comment by RowanMaBoot** [comment]
Apologies for my complete hiatus as I have been very busy over the last few months with work (had a rather large promotion), so I've barely touched my computer (my SO is also eating up most of my free time, so I rarely touch my PC right now - and when I do, it is because she wants to play or see something).

@TitanRogue

I'd say we should go with what Tanner said as it makes updating easier (when I was porting events over again, I was avoiding changing var names etc for the sake of it, makes using merge/compare tools possible as well).

As for the steam page I'm not sure how to as you're added as a project owner on the steam page, afaik. So I believe you should be able to update it? If not we can look at alternatives.

Regardless, I'll get it updated tonight -- tomorrow at the latest.

**Comment by TitanRogue** [comment]
@Altemio Unfortunately the Steam Workshop is very limited when it comes to cooperative projects. So I would make a reupload and link the new page in the comments and the description of the old page would also have a link to the new one.

I will then use the government that is referenced for the protectorate. I believe it is a feudal government but I could easily make it a republican one. That is then the only thing we should think about.
I also found that the Dothraki and Rhoynish invasion have been deleted. I will bring them back.

Congrats for your promotion.

**Comment by TitanRogue** [comment]
There is a huge game breaking issue with one of my uploads yesterday. I fixed it but it is very important that you update the mod. The issue makes the game freeze after a few months and makes it impossible to play.

**Comment by TitanRogue** [comment]
@Altemio 
What do you think of my suggestion?

**Comment by RowanMaBoot** [comment]
We can do that if you want! Though we'll lose our subscribers & ratings, if that matters at all.

Mind I'm active again (and will be for the foreseeable future, since I'm also not planning on moving again either) - so worst case you can drop a message here (@ signing me) and I'll be able to priority upload any new builds as needed.

**Comment by TitanRogue** [comment]
Meant no offense. Did not know you where active again. Then we can leave it the way it was, I guess.

**Comment by TitanRogue** [comment]
I think this is Beta4 now, isn't it? Also someone should take care of the changelog and the mod description.

**Comment by TitanRogue** [comment]
![thumbnail](https://user-images.githubusercontent.com/40209082/66406131-6ace4480-e9eb-11e9-9cd3-c13ce585d7b2.png)
I also made a new thumbnail inspired by the old one.

**Comment by RowanMaBoot** [comment]
Oh no, I didn't take any offense!! I'm totally fine with it if that is the preferred option, wouldn't have had any issues if you guys created a new page, especially whilst I was gone (I was completely unaware that you guys were unable to upload).

I'll get onto version related stuff if someone doesn't beat me to it.

And nice! Looks great - I'll get that thumbnail added once I get home.

**Comment by TitanRogue** [comment]
I am currently assigning a terrain type to every province, I hope you guys don't mind.

**Comment by TitanRogue** [comment]
Westerosi Feudalism should be renamed to Andal Feudalism since actually only the Seven rulers get it. They are kinda ruling in the name of the gods whilst the First Men following the Old Gods have a more martial and powerbased feudal or tribal structure just like pagan Northmen. What do you think.

**Comment by tannerflick4** [comment]
Sounds good to me. While old gods rulers should still follow traditional succession, wildlings could probably get some type of succession similar to the Byz in vanilla, where the strongest "commander
" becomes the next ruler if theres no heirs

**Comment by RowanMaBoot** [comment]
Yeah, I think it was Turkish succession? If so that would suit the Wildlings really well!

**Comment by tannerflick4** [comment]
@Altemio , I noticed HF 23005 we have twice in HF religious events, because we added an extra set of options from 23004. But, see the next event is same id, so run into a problem. But I would rather you fix that.

**Comment by RowanMaBoot** [comment]
Should be fixed now!

**Comment by tannerflick4** [comment]
Why do we have our own generate spouses rule? Shouldn't we just use the vanilla generate families rule?

**Comment by TitanRogue** [comment]
The diffrence is that the Vanilla rule only adds families for lonely rulers without wife and children. Since we have rulers with children but no spouse I made another rule to fill that gap.

**Comment by tannerflick4** [comment]
Sounds good, good work!

**Comment by TitanRogue** [comment]
The carrion pass should be another wasteland. The prince's pass and the bone way are historically the only passable ways to Dorne.

**Comment by TitanRogue** [comment]
Oh and how do you feel about CKIII?

**Comment by tannerflick4** [comment]
I like ck3! I enjoy the roleplay/character focus, but worry modding wise it will resemble I:R too much which I dislike the way it throws everything into a few files.

**Comment by TitanRogue** [comment]
@Altemio 
Could you update the mod?
I made some fixes for your new infrastructure decisions and allowed rulers to use them in their vassal provinces, I hope you are fine with that.

**Comment by RowanMaBoot** [comment]
@TitanRogue

Sure! 

And that is absolutely no problem, threw that one together on my phone during lunch - I'm actually focusing on getting the building system working flawlessly, and the road & infrastructure modifiers are going to be playing a fairly large part in that, so that is great.

**Comment by tannerflick4** [comment]
@Altemio , can you update the mod again? We fixed a critical bug. thanks, and sorry to bother ya. :)

**Comment by RowanMaBoot** [comment]
No problem! Checking every night now to see if we need to upload a new build xD

**Comment by TitanRogue** [comment]
@Altemio 
Could you alter the recommended mods section of the mod's description to
- DLC Portraits: https://steamcommunity.com/sharedfiles/filedetails/?id=1360765985
- In heaven's graphical overhaul compatibility: https://steamcommunity.com/sharedfiles/filedetails/?id=1880960100
- Dragon Engine: https://steamcommunity.com/sharedfiles/filedetails/?id=1880024189
?
Could you also link our Discord server?

**Comment by RowanMaBoot** [comment]
Forgot to say, but got that sorted - all of those details have been popped into the description 👍

**Comment by TitanRogue** [comment]
@Altemio 
We fixed lots of bugs. Would be nice if you could update the mod. ;)

**Comment by RowanMaBoot** [comment]
Getting it uploaded now :)

**Comment by TitanRogue** [comment]
@Altemio 
Could you please remove
(Outdated) CPR+ for TSK: http://steamcommunity.com/sharedfiles/filedetails/?id=1360765985.
from the description?

**Comment by TitanRogue** [comment]
@Altemio 
The mod descrition says that the current version is beta5 but it actually is beta6. Since we already have the version's name in the mod's name I think we can delete it from the description.

**Comment by RowanMaBoot** [comment]
Yeah that is a good point; I've dropped that from the description. 👍

**Comment by TitanRogue** [comment]
@Altemio  @tannerflick4 
Is there any system behind the localisation files naming? If not I would like to give them all a 000_ prefix except mapfiller files.

**Comment by tannerflick4** [comment]
Not that I know of. Just kinda whatever goes lol.

**Comment by TitanRogue** [comment]
@Altemio  @tannerflick4 
Unfortunately I can not add custom terrain defender bonuses so only vanilla terrain bonuses work. I suggest that we delete the new terrain types in order to assure strategic meaning of terrain.

Edit: I have found a way to keep the new terrains with bonuses but I will need some time to balance and adjust it.

**Comment by Vierwood** [comment]
@Altemio someone might've found a fix for our straight crossing issue. 

https://forum.paradoxplaza.com/forum/index.php?threads/strait-crossing-graphics-not-showing-up.1276114/

**Comment by TitanRogue** [comment]
Could work. I can take a look at it in a few hours if you do not mind.

**Comment by RowanMaBoot** [comment]
Ooo. Great find! 👍

**Comment by TitanRogue** [comment]
I made some changes and we now have visible straits.

**Comment by RowanMaBoot** [comment]
So with the new buildings, the overall levy size is being drastically reduced.. It will range from, as a completely un-upgraded castle, roughly 85 men, to approximately 1000 men once fully upgraded.

This should help with certain aspects of the economy, notably being:

- Less men, which means less economic drain during most conflicts. This also means mercenaries will be more impactful, as they will make a bigger difference in regards to the number of men fielded.
- Smaller garrisons, due to a smaller (raw) number of troops.
- Doomstacks will be a lot smaller, and less prone to attrition without changing how attrition works.
- Smaller forces means for the potential for more granular military control too, although I expect that (generally speaking) doomstacks will be fairly dominant, most wars will fall below the attrition limits rather than massively exceeding them.

Retinue size is also being re-introduced for a number of buildings, and I'll be adding the ability to put together a retinue. It'll be very small, ranging from a few dozen nights. With smaller levy sizes, a 32 Knight retinue for a small duke to a 500 man retinue as a King will actually be impactful and hopefully somewhat loreful.

This will help balance lords, since we can look to buffing significant lords - I'll be assigning, dynamically, infrastructure modifiers to Kings and Dukes, which will give them considerably better starting development. That means we might be able to revert some of the faction neutering Vierwood did, so that factions actually play a role (but two dukes out of eight won't immediately overpower the King at the beginning in regards to levy and economic strength).

**Comment by RowanMaBoot** [comment]
So the 00_core_building_setup will be the file that governs building assignment at game start. It assigns based on region and rank, with considerably better infrastructure given to capital versus secondary demesne.

I'll add a secondary event for forcing infrastructure & buildings, if we then want particular provinces (or realms/duchies/kingdoms etc) to be even more developed.

If you guys have any suggestions or questions, lemme know. The idea is that we shall use it for managing buildings (and means no more editing history files to add buildings, which'll make thing a lot less painful and it'll make it easier to see what has what against what, since it will all be in one place) and the sort.

**Comment by RowanMaBoot** [comment]
Doing some last minute testing and scripting, but I've got 30,000 lines of building code almost ready to commit. Will roll out a modified development event chain at around the same time so that buildings are populated globally (based on the rank and location of the character).

I've written some python tools for helping with generating buildings, so I can do bulk edits/creates now. So let me know if anyone has ideas for buildings that could be added and what they might rely on (resources, terrain etc).

**Comment by tannerflick4** [comment]
With the latest commit (my parse fixes), the game crashed for me after a year or 2, playing in the time of the Hammer of the Dornish. error log was 33000 lines :(

**Comment by RowanMaBoot** [comment]
Ah, not so fabulous. Will get that checked out.

**Comment by RowanMaBoot** [comment]
Very strange, I'm able to play for decades without any issues. But I'm playing as an Old Gods character, Rain House. Going through and squashing errors right now, trying to get the error log down - will then playtest the Hammer. Can you replicate the crash easily as him, but not as say Old Gods characters?

**Comment by tannerflick4** [comment]
I was playing as a count in the Crag as an Old Gods. Weird :(

**Comment by tannerflick4** [comment]
@Altemio , build times for buildings are coming out with not straight up days, and I think the game only allows/prefers "164" instead of "164.28"...

**Comment by RowanMaBoot** [comment]
Oops! Wasn't aware it didn't like decimal values. I'll make sure to fix that. I've been tweaking and rebalancing everything, so will make sure that is remedied.

**Comment by tannerflick4** [comment]
I really want to get the mod to not crash at any point. For me, it always crashes from 2-40 years in, usually earlier.

But, a lot of things I cannot fix on my own. Bringing the EMF stuff up because I don't know why it was ever added, and its code is too complex for me to figure out what stays, what should go, what is actually used, etc.

**Comment by TitanRogue** [comment]
I am currently working on a bug fix for the EMF coronation system combined with improvements for creating king-tier titles when ruling an empire.
I have also tested my map editing skills and I came up with this. I simply moved the Codd incest isle northwards and made it smaller. That island always felt out of place to me. If you agree I commit it and improve it.
![20200428155933_1](https://user-images.githubusercontent.com/40209082/80496640-c6512700-8969-11ea-8fbf-e423b83c9a56.jpg)

**Comment by tannerflick4** [comment]
@Altemio , there are many duplicate knight modifiers in the terrain bonuses/maluses. The numbers are different for them, so you'll have to edit that ;)

**Comment by RowanMaBoot** [comment]
I'm not too sure why so much EMF stuff was brought in - mechanically it's a sweet mod, and it'd be nice to have them integrated. But checking a few of the events added, they weren't adapted at all for our mod and still reference vanilla/HIP content!

So for the content that hasn't been adapted, we'll defo want to look into making adjustments.

Since you're getting crashing so frequently @tannerflick4, could you try messing around with a couple of things, time permitted of course, to see if crashing is eliminated or reduced for you?

1. Remove stuff like decisions, EMF events, resource system and the sort. See if you can determine if there is a particular trigger crash-wise in those?
2. The map also appears to hammer my Graphics Card/GPU. When I zoom in my GPU fan accelerates and usage goes up, whilst zooming out has the opposite effect. Kinda unusual, so makes me wonder if some users are crashing due to GPU behaviour. So if you wanted to remove the altas0 files in map/terrain/, remove the terrains.txt in /map/ and then delete the entire /gfx/FX folder - could you see if that impacts crashing for you?

**Comment by TitanRogue** [comment]
@Altemio 
I do believe the tree map is too big and as a consequence causes massive lags. When zooming in too many trees are generated. That's what I believe.
Some EMF triggers and effects are used by the HIP factions. I have started checking them out and will continue to do so. I also try to fix Improved Genetics.

**Comment by tannerflick4** [comment]
@Altemio , I tried removing those files but now the mod does not load for me. It crashes after the first loading screen loads.

**Comment by RowanMaBoot** [comment]
@TitanRogue 
That actually makes sense and could be the cause! I've not had any lag when zooming in, but that could also just be my system. Will defo want to explore vanilla tree settings, see if that has bearing on the instability. I'm not sure it'd be a root cause in this case however, as I'm fairly sure HIP also does similar (altho maybe not quite as high tree density).

@tannerflick4 
In the case of The_Seven_Kingdoms\gfx\FX, delete everything excluding the trees.lua file. If it continues to crash after that, then try also keeping the original terrains.txt file.

**Comment by TitanRogue** [comment]
I resized the tree map in my latest commitment. I just forgot to note it.

**Comment by tannerflick4** [comment]
Leaving the terrains file in but removing the others gets me to about 30 years in game, still not great, but better than 10-15 with the files included.

**Comment by RowanMaBoot** [comment]
Very bizarre - thanks for experimenting! Let me know how events impact the instability, if by much at all.

**Comment by TitanRogue** [comment]
@Altemio @tannerflick4 
The reason for the mod loading and then crashing when removing terrain.txt is terrain.bmp since terrain.txt is referencing to new terrain types in terrain.bmp. The vanilla terrain.txt doesn't have those references making those added colours in terrain.bmp "illegal" and then casues a crash, I guess.

**Comment by RowanMaBoot** [comment]
Yeah - when I made the original suggestion to remove the terrain.txt file it completely passed me by, as the indexes would need to be stripped from the terrain file, so it's just easier to keep that file. The extra terrain types and the larger trees.bmp file shouldn't have any bearing on the issues seen - just because many other mods do the same without trouble. Still was worth a try tho.

Another thing we'll need to look at is making sure all islands are defined as such, as the AI itself may be breaking somewhat. Not sure how dependent the game is on island definitions, and what might be going on in that regard.

**Comment by tannerflick4** [comment]
I am sure as we continue to fix up things it could get better. 

Titan is going to hammer out the EMF files, that will help too.

**Comment by tannerflick4** [comment]
Helios: Hello there, hope you don't mind me further explaining that problem I ran into over here.
[9:33 AM] Helios: Okay, so lets take Oldtown as an example, where the Starry Sept is mine and not the high septon's. So in order to build the small trivial vineyards in Oldtown I need minor chapel V, however even when the Starry Sept has minor chapel V, I can only build those vineyards in the Starry Sept, not in Oldtown as I wanted. Perhaps because the required building has to be in the same sub-holding/capital as the building you want to build?
[9:34 AM] tsf4: yeah, thats how ck2 buildings work. you'd need the minor chapel V for oldtown to get the vineyards there
[9:41 AM] Helios: But erhm, there is no minor chapel V building buildable in Oldtown. Closest thing you can build in a castle-holding is the altar line of buildings, which is maxed at lvl III. Same goes for goat/sheep pastures and the lvl III shrine requirement, which also can't be build in castle-type holdings.
[9:43 AM] tsf4: ok, i'll note this to Rowan, he has been working on buildings

Had this exchange with a dude on Discord

**Comment by RowanMaBoot** [comment]
Great to know! It's a bit odd that a 'Chapel V' building would be referenced by a building, given that such doesn't exist (or shouldn't). Was the user using the Git version? If so, did they do a clean install when using the latest release? I'll look into it nevertheless, may already be fixed (I'll try to get the building overhaul stuff committed asap).

**Comment by tannerflick4** [comment]
Currently, it doesn't seem possible to implement absolute cognatic succession. I've tested a few Free Folk cultures as well as a Valyrian and Dornish and none of them are considered to meet the condition of being in their culture groups.

This was reported by someone to the Discord who plays the mod through steam. I tried adding the culture groups to the potential block, but it still didn't work/was not available for them to select/greyed out.

**Comment by RowanMaBoot** [comment]
That should be fixed as of my last commit, the succession laws seemed to have a lot of issues in general for some reason. Let me know if we see any further weirdness (hopefully it should also fix Valyrian succession, as people were reporting that broke)

Now the important bit - crashes!

- When do we see crashes? What is the average amount of time that spans before seeing a crash?
- Do the crashes occur more frequently when playing as certain characters or cultures?
- Do crashes occur when using observe mode. Is it more or less frequent?
- Do the crashes occur when events or decisions are present? If we strip them from the mod, does it affect the presence of said crashes?

I'm doing testing on my end to see if we can get to the bottom of this right now. If any of the above questions can be answered tho, would be incredibly helpful.

Edit:

Defo getting crashes when events and decisions have been stripped, and still seeing crashes via observe or when playing as a character.

**Comment by RowanMaBoot** [comment]
Strangely enough I think I might've found the cause!

**Comment by tannerflick4** [comment]
Made a new post for us to all comment on, this one is too long and my internet grumbles loading it.
